pkg/build/builder/docker.go

```diff
@@ -70,40 +66,27 @@ func (d *DockerBuilder) Build() error {
 		return err
 	}
 	glog.V(4).Infof("Starting Docker build from %s/%s BuildConfig ...", d.build.Namespace, d.build.Name)
-	var push bool
-
-	// if there is no output target, set one up so the docker build logic
-	// will still work, but we won't push it at the end.
-	if d.build.Spec.Output.To == nil || len(d.build.Spec.Output.To.Name) == 0 {
-		d.build.Spec.Output.To = &kapi.ObjectReference{
-			Kind: "DockerImage",
-			Name: noOutputDefaultTag,
```

Comment:
@rhcarvalho the problem is that you need to provide a tag, even if you're not going to push, otherwise the docker build will fail, that's why i added the code that provided the dummy value in the first place.